components/identities/availableIdentities.tsx

```diff
@@ -1,272 +1,225 @@
-        })
-        .then((data: IdentityData) => {
-          if (minting) endMinting();
-          setIdentity(new Identity(data));
-          setIsIdentityADomain(Boolean(data?.domain));
-        })
-        .catch(() => {
-          // Domain data might not be indexed yet, so we check local storage
-          const domainData = getDomainData(tokenId);
-          if (domainData) {
-            setIdentity(new Identity(domainData));
-            setIsIdentityADomain(Boolean(domainData?.domain));
-          } else {
-            setIsIdentityADomain(false);
-          }
-        }),
-    [tokenId, minting, endMinting]
-  );
-
-  useEffect(() => {
-    if (minting && tokenId && !identity) {
-      const interval = setInterval(() => refreshData(), 1000);
-      return () => clearInterval(interval);
-    }
-  }, [minting, tokenId, identity, refreshData]);
-
-  useEffect(() => {
-    if (tokenId) {
-      refreshData();
-      const timer = setInterval(() => refreshData(), 30e3);
-      return () => clearInterval(timer);
-    }
-  }, [tokenId, refreshData]);
-
-  function mint() {
-    execute();
-  }
-  useEffect(() => {
-    if (address) {
-      // Our Indexer
-      fetch(
-        `${
-          process.env.NEXT_PUBLIC_SERVER_LINK
-        }/addr_to_full_ids?addr=${hexToDecimal(address)}`
-      )
-        .then((response) => response.json())
-        .then((data) => {
-          setOwnedIdentities(data.full_ids);
-        });
-    }
-  }, [address, router.asPath]);
+ 
+   useEffect(() => {
+      if (isIdentityADomain === true) {
+         setHideActions(true);
+      } else {
+         setHideActions(false);
+      }
+   }, [isIdentityADomain]);
+
+   const refreshData = useCallback(
+      () =>
+         fetch(`${process.env.NEXT_PUBLIC_SERVER_LINK}/id_to_data?id=${tokenId}`)
+            .then(async (response) => {
+               if (!response.ok) {
+                  throw new Error(await response.text());
+               }
+               return response.json();
+            })
+            .then((data: IdentityData) => {
+               if (minting) endMinting();
+               setIdentity(new Identity(data));
+               setIsIdentityADomain(Boolean(data?.domain));
+            })
+            .catch(() => {
+               // Domain data might not be indexed yet, so we check local storage
+               const domainData = getDomainData(tokenId);
+               if (domainData) {
+                  setIdentity(new Identity(domainData));
+                  setIsIdentityADomain(Boolean(domainData?.domain));
+               } else {
+                  setIsIdentityADomain(false);
+               }
+            }),
+      [tokenId, minting, endMinting]
+   );
+
+   useEffect(() => {
+      if (minting && tokenId && !identity) {
+         const interval = setInterval(() => refreshData(), 1000);
+         return () => clearInterval(interval);
+      }
+   }, [minting, tokenId, identity, refreshData]);
 
```

Comment:
You can simplify this logic by just doing `setHideActions(!isIdentityADomain)` or `setShowActions(isIdentityADomain)`